Week 04/Lecture 08/Assignment 2/src/main/java/jebi/hendardi/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,65 @@
+package jebi.hendardi.repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import jebi.hendardi.model.Employee;
+
+@Repository
+public class EmployeeRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public EmployeeRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Employee> EMPLOYEE_ROW_MAPPER = new RowMapper<>() {
+        @Override
+        public Employee mapRow(ResultSet rs, int rowNum) throws SQLException {
```

Comment:
mapRow shoud be separated into another class, just for mapping data

Reply:
The RowMapper logic has been moved to a separate class named EmployeeRowMapper in directory /mapper. This new class is responsible solely for mapping ResultSet data to the Employee entity, adhering to the single responsibility principle.

**Changes Made:**
- Created EmployeeRowMapper.java for handling the mapping.
- Updated EmployeeRepository.java to use EmployeeRowMapper.